web_src/js/components/RepoActionView.vue

```diff
@@ -193,6 +197,11 @@ const sfc = {
       await this.fetchPost(`${jobLink}/rerun`);
       window.location.href = jobLink;
     },
+    // rerun workflow
+    rerun() {
+      this.fetchPost(`${this.run.link}/rerun`);
+      window.location.href = `${this.run.link}`;
```

Comment:
```suggestion
      window.location.href = this.run.link;
```

Reply:
And here is a common frontend bug: it needs "await" the ajax response before "reload page".

Reply:
There is [require-await](https://eslint.org/docs/latest/rules/require-await) lint rule, might be worth checking out.

Reply:
Technically this could be linted for, but so far no eslint plugin has implemented a usable version of this. See also https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1144 for discussion.

Reply:
https://github.com/go-gitea/gitea/pull/24350/files#r1181080428